packages/trader/src/Stores/Modules/Trading/Helpers/contract-type.ts

```diff
@@ -612,7 +611,7 @@ export const ContractType = (() => {
         const arr_cancellation_range: string[] =
             getPropertyValue(available_contract_types, [contract_type, 'config', 'cancellation_range']) || [];
 
-        const regex = new RegExp('^([0-9]+)|([a-zA-Z]+)$', 'g');
+        const regex = /^\d{1,}|[a-zA-Z]{1,}$/g;
```

Comment:
1. used a regular expression literal instead of the 'RegExp' constructor.
2. replaced + with {1,} equivalent in order to prevent the regex from being vulnerable to denial of service (DoS) attacks due to backtracking.
3. used concise character class syntax '\d' instead of '[0-9]'.
4. removed extra brackets around 2 OR operands in order to make the intended operator precedence explicit.